utils/api.ts

```diff
@@ -73,86 +75,58 @@ export function setHooks(hooks: {
   set(refresh, hooks.refresh);
 }
 
-export function sleep(ms = 0, signal: AbortSignal): Promise<void> {
-  return new Promise((resolve, reject) => {
-    const id = setTimeout(() => resolve(), ms);
-    signal?.addEventListener('abort', () => {
-      clearTimeout(id);
-      reject(new Error('request aborted'));
-    });
-  });
-}
-
-export async function fetchWithCsrf<
-  Resp,
-  Req extends NitroFetchRequest = NitroFetchRequest,
->(request: Req, options?: NitroFetchOptions<Req>) {
-  const { baseUrl } = useRuntimeConfig().public;
+export const fetchWithCsrf = $fetch.create({
+  credentials: 'include',
+  async onRequest({ options }) {
+    const { baseUrl } = useRuntimeConfig().public;
+    const event = typeof useEvent === 'function' ? useEvent() : null;
 
-  let token = useCookie(CSRF_COOKIE).value;
+    let token = event
+      ? parseCookies(event)[CSRF_COOKIE]
+      : useCookie(CSRF_COOKIE).value;
 
-  if (
-    process.client
-    && ['post', 'delete', 'put', 'patch'].includes(
-      options?.method?.toLowerCase() ?? '',
-    )
-  ) {
-    await initCsrf();
-    token = getCookie(CSRF_COOKIE);
-  }
+    if (
+      process.client
+      && ['post', 'delete', 'put', 'patch'].includes(
+        options?.method?.toLowerCase() ?? '',
+      )
+    ) {
+      await initCsrf();
+      token = getCookie(CSRF_COOKIE);
```

Comment:
the `initCsrf()` seems to return the value of the `CSRF_COOKIE`, can we use the value from there instead and remove the `getCookie()` part?

Reply:
yes we can that was the intent but I forgot :)